src/utils/handler.js

```diff
@@ -1,16 +1,140 @@
-export const addToWishListHandler = async (item) => {
-    try {
-      const config = {
-        headers: {
-          authorization: localStorage.getItem("token"),
-        },
-      };
-      const data = {
-        product: item,
-      };
-      await axios.post("/api/user/wishlist", data, config);
-
-    } catch (err) {
-      console.log("err:- " + err);
-    }
-  };
+import axios from "axios";
+import {
+  WISHLIST_LOADER,
+  CART_DATA,
+  WISHLIST_DATA,
+  CART_INCREMENT,
+  CART_DECREMENT,
+  TOTAL_PRICE
+} from "../reducer/wishlist/wishlistConstants";
+import {
+  PRODUCT_LIST_DATA,
+  LOADING_SPINNER,
+} from "../reducer/productList/productConstants";
+import { USER_LOADING, LOGIN_STATUS } from "../reducer/user/userConstants";
+
+const encodedtoken = localStorage.getItem("token");
+const config = {
+  headers: {
+    authorization: encodedtoken,
+  },
+};
+
+
+// Auth
+export const login = async (email, password, dispatch, navigate) => {
+  try {
+    const response = await axios.post("/api/auth/login", {
+      email,
+      password,
+    });
+    localStorage.setItem("token", response.data.encodedToken);
+    navigate("/");
+    dispatch({ type: USER_LOADING });
+    dispatch({type: LOGIN_STATUS})
+    setTimeout(() => dispatch({ type: USER_LOADING }), 1000);
+  } catch (err) {
+    console.log("Error: ", err);
+  }
+};
+
+
+// Product List
+export const getProductList = async (dispatch) => {
```

Comment:
every time for repeating url such as /api/ at top define in axios.defaults.baseUrl="/api/" then no need to repeat in every axios call (just a suggestion)